functions/Invoke-DbaDiagnosticQuery.ps1

```diff
@@ -1,94 +1,143 @@
+    .EXAMPLE
+    Invoke-DbaDiagnosticQuery -SqlInstance sql2016
 
-Then it will export the results to Export-DbaDiagnosticQuery.
+    Run the selection made by the user on the Sql Server instance specified.
 
-#>
+    .EXAMPLE
+    Invoke-DbaDiagnosticQuery -SqlInstance sql2016 -UseSelectionHelper | Export-DbaDiagnosticQuery -Path C:\temp\gboutput
+
+    Provides a gridview with all the queries to choose from and will run the selection made by the user on the SQL Server instance specified.
+    Then it will export the results to Export-DbaDiagnosticQuery.
+
+    .Example
+    # Exporting Queries To SQL Files
+    - Parse the appropriate diagnostic queries by connecting to server to get version matched queries
+    - Instead of running the diagnostic queries, export them to SQL files
+
+    # Export All Queries to Disk
+    Invoke-DbaDiagnosticQuery -sqlinstance localhost -ExportQueries -outputpath "C:\temp\DiagnosticQueries"
+
+    # Export Database Specific Queries for all User Dbs
+    Invoke-DbaDiagnosticQuery -sqlinstance localhost -DatabaseSpecific -DatabaseName 'tempdb' -ExportQueries -outputpath "C:\temp\DiagnosticQueries"
+
+    # Export Database Specific Queries For One Target Database
+    Invoke-DbaDiagnosticQuery -sqlinstance localhost -DatabaseSpecific -DatabaseName 'tempdb' -ExportQueries -outputpath "C:\temp\DiagnosticQueries"
+
+    # Export Database Specific Queries For One Target Database and One Specific Query
+    Invoke-DbaDiagnosticQuery -sqlinstance localhost -DatabaseSpecific -DatabaseName 'tempdb' -ExportQueries -outputpath "C:\temp\DiagnosticQueries" -queryname 'Database-scoped Configurations'
+
+    # Choose Queries To Export
+    Invoke-DbaDiagnosticQuery -sqlinstance localhost -UseSelectionHelper
+
+    This will export with SqlInstance, DatabaseName, and QueryName as appropriate based on query.
+
+    .Example
+    # Export Queries (not run) as returned object
+
+    [System.Management.Automation.PSObject[]]$results = Invoke-DbaDiagnosticQuery -SqlInstance localhost -whatif
+
+    Parse the appropriate diagnostic queries by connecting to server, and instead of running them, return as [pscustomobject[]] to work with further
+
+    .Example
+    # Database Specific Handling
+    $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific -queryname 'Database-scoped Configurations' -databasename $database
+
+    Run diagnostic queries targeted at specific database, and only run database level queries against this database.
+
+    #>
 
     [CmdletBinding(SupportsShouldProcess)]
+    [outputtype([pscustomobject[]])]
     param (
         [parameter(Mandatory = $true, ValueFromPipeline = $true, Position = 0)]
         [Alias("ServerInstance", "SqlServer")]
         [DbaInstanceParameter[]]$SqlInstance,
+
+        [Alias('Database')]
+        [string[]]$DatabaseName,
```

Comment:
This needs to be changed to `[object[]]$Database` so the TEPP will kick in properly. You can remove the alias.

Reply:
Our standard is to also include `[object[]]$ExcludeDatabase` to allow folks to have the opposite filter.

Reply:
I didn't remove Database alias, as that was the previous name, wanted to prevent breaking existing scripts, so aligned name with standard, but left alias to allow database to be used.
The exclude is a good idea, so I'll include in the check-in.

Reply:
go the other way around. TEPP will kick in only on `Database` and `ExcludeDatabase`. Every existing dbatools function with `DatabaseName` as a parameter will be changed to use `Database` (optionally using `DatabaseName` as the alias)